lib/bolt/result_set.rb

```diff
@@ -106,5 +106,13 @@ def to_s
     def ==(other)
       eql?(other)
     end
+
+    def [](from, up_to = nil)
+      if up_to
+        @results[from, up_to]
```

Comment:
While this is the right syntax for Puppet, it looks like for Ruby this doesn't support negative indexing (which the Puppet one does support :P). I think making this `@results[from..up_to]` will work as the Puppet one does, which is what I think users will expect.

Example:
```
out::message("${$result[1, -1]}")
```

Reply:
wow, good catch. Thanks!